cobol-parser/src/main/scala/za/co/absa/cobrix/cobol/parser/decoders/StringDecoders.scala

```diff
@@ -54,7 +70,15 @@ object StringDecoders {
         buf.append(bytes(i).toChar)
       i = i + 1
     }
-    buf.toString.trim
+    if (trimmingType == TrimNone) {
+      buf.toString
+    } else if (trimmingType == TrimLeft) {
+      StringTools.trimLeft(buf.toString)
+    } else if (trimmingType == TrimRight) {
+      StringTools.trimRight(buf.toString)
+    } else {
+      buf.toString.trim
```

Comment:
maybe a Scala `match`? Of course, in case this method is not critical for performance.

Reply:
yes, here it is also critical to performance. :)
That's why I even used numeric constants instead of Enumerable or case objects.